### DISCUSSION TITLE
Add-on to Product

### DISCUSSION THREAD

**Discussion by imash96** [medusajs/medusa#5001] [Feature Requests]
Hi,

I loved playing with medusa while building a storefront alongside a customized medusa backed. I came across a use case like I have a product like T-shirt so can I provide buyer with add-on service like custom image printing or buyer name printing on the product or like we order pizza we can an option to addon toppings and extra amount get added to the cart based on addon selection.

Is there already any way we can implement this or can you suggest how I can implement this feature to my Medusa backend.

I think this feature will be very helpful to many of the medusa user as many products have addon to offer to there customer.

Thank you

**Comment by sittingbool** [comment]
Hey, we just had this issue and so in pur team tried to solve that but it broke compatibility and was very much a hazzle. Can this be resolved using a plugin or would that have negative impact?

**Comment by WildsG** [comment]
I've also encountered this problem, where it's very hard to add such functionality to already existing medusa flow. Does anybody has a solution?

**Reply by SianAli24** [reply]
Were you able to figure it out? @WildsG

**Comment by SianAli24** [comment]
Did you achieve it? @imash96
